src/MainWindow.cpp

```diff
@@ -1038,6 +1039,36 @@ void MainWindow::wantExitFilesManagement()
     updateTabButtons();
 }
 
+void MainWindow::displayLoginRequestMessageBox(const QString& message)
+{
+    QJsonParseError err;
+    QJsonDocument jdoc = QJsonDocument::fromJson(message.toUtf8(), &err);
+    if (err.error != QJsonParseError::NoError)
+    {
+        qWarning() << "JSON parse error " << err.errorString();
+        return;
+    }
+    QJsonObject rootobj = jdoc.object();
+    QJsonObject o = rootobj["data"].toObject();
```

Comment:
All the json decoding should stay in WSClient. It's part of the json/websocket procotol, it should not be done in mainwindow

Reply:
moved to WSClient